roles/rsyslog/roles/output_roles/elasticsearch/tasks/main.yaml

```diff
@@ -29,34 +29,35 @@
         - '{{ rsyslog_elasticsearch }}'
       when:
         - item.key_src is defined
-        - item.key_src != ""
+        - item.key_src
 
     - name: Update ca_cert path to /etc/rsyslog.d directory in Rsyslog
       set_fact:
-        rsyslog_elasticsearch_temp: "{{ rsyslog_elasticsearch_temp | d([]) }} + {{ [ item | combine( { 'ca_cert': item.ca_cert_src | basename , 'cert': item.cert_src | basename, 'key': item.key_src | basename } ) ] }}"
+        __rsyslog_elasticsearch_temp: >-
+          {{ __rsyslog_elasticsearch_temp | d([]) }} +
+          {{ [ item | combine( { 'ca_cert': item.ca_cert_src | basename ,
+          'cert': item.cert_src | basename, 'key': item.key_src | basename } ) ] }}
       with_items:
         - '{{ rsyslog_elasticsearch }}'
       when:
         - item.ca_cert_src is defined
-        - item.ca_cert_src != ""
+        - item.ca_cert_src
```

Comment:
@nhosoi do they both mean the same
 item.ca_cert_src vs. item.ca_cert_src != ""   ?

Reply:
Yes.  This is part of adapting the ansible system roles best practices.

Reply:
@nhosoi are you sure? Also which part of the best practices are you referring to?

Reply:
see https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_2.8.html#bare-variables-in-conditionals

Reply:
@sradco, @pcahyna, sorry, I obviously made a mistake.  Reverting this line.

Reply:
@sradco I am afraid they do not mean the same - this is a complicated subject, see the discussion in https://github.com/ansible/ansible-lint/issues/457 and related issues. Bare variables used to be treated differently than expressions in conditionals. I.e. `something` is not the same as `not (not something)`, even if evaluated in a boolean context.

Reply:
@nhosoi I think we should disable the problematic lint rule (E602), but there should be no problem with using `| length > 0` as well. Actually, I think now that `item.ca_cert_src` was actually correct, so @sradco was right - because it is not a bare variable - the dot in the middle makes it an expression. This is all so confusing :-(

Reply:
@nhosoi by the way, I just disabled the lint rule 602, which disallows comparisons to empty string, globally in the template. https://github.com/linux-system-roles/template/pull/19 . You can safely use comparison to empty string when you believe it is the best option. (After this change is merged.)

Reply:
Thank you for the update, @pcahyna !